packages/shared_preferences/lib/shared_preferences.dart

```diff
@@ -60,7 +64,17 @@ class SharedPreferences {
 
   /// Reads a set of string values from persistent storage,
   /// throwing an exception if it's not a string set.
-  List<String> getStringList(String key) => _preferenceCache[key] as List<String>;
+  List<String> getStringList(String key) =>
+      _preferenceCache[key] as List<String>;
+
+  /// Gets a JSON-encoded preference string and returns it as an Object. See
+  /// [JSON.decode] and [JsonDecoder] documentation for what happens when the
+  /// string is not a valid JSON string, and how to use the reviver.  Returns
+  /// null if the given key does not exist.
+  Object getObject(String key, {reviver(var key, var value)}) {
+    if (_preferenceCache[key] == null) return null;
```

Comment:
I do see what you mean. I guess I was mostly thinking of this as a way to store small structured data objects, but I can see that it could easily get out of hand if you store a large tree of data.

The other way I thought to avoid some encoding, was to do it on the native side, similar to the way the string lists are stored, so perhaps that's a better tack to take there.  Of course then it has to be done twice, and differently for each native implementation.

So, given that it's meant to be synchronous, that pretty much precludes _any_ kind of nested data, since you don't want to allow a single large item to slow things down, right?  Would it make sense to have something that provides an async API for loading these things?

As far as proceeding, given the synchronous constraint, I think I'll basically abandon this change, except for adding the assert.